lite2/client.go

```diff
@@ -728,16 +733,21 @@ func (c *Client) sequence(
 		}
 
 		// Update trusted header and vals.
+		trustedHeader = interimHeader
 		if height == newHeader.Height-1 {
-			interimNextVals = newVals
+			trustedNextVals = newVals
 		} else {
-			interimNextVals, err = c.validatorSetFromPrimary(height + 1)
+			trustedNextVals, err = c.validatorSetFromPrimary(height + 1)
 			if err != nil {
 				return errors.Wrapf(err, "failed to obtain the vals #%d", height+1)
 			}
+			if !bytes.Equal(trustedHeader.NextValidatorsHash, trustedNextVals.Hash()) {
```

Comment:
probably a bad idea to set something as trusted `trustedNextVals` unless we can trust it